spec/parser/ruby/ruby_parser_spec.rb

```diff
@@ -283,6 +283,58 @@ def hello; end
       end
     end
 
+    it "\n     Parses several formats of percent arrays into source strings\n" \
+         "     in constant declarations.  Loops thru all combinations of\n" \
+         "     () {} [] <> || delimiters, %i %I %w %W array types, and five\n" \
+         "     types of element/delimiter whitespace." do
+      delimiter_h = {
+        '(' => ')',
+        '{' => '}',
+        '[' => ']',
+        '<' => '>',
+        '|' => '|'
+      }
+
+      type_h = {
+        'i' => :qsymbols_literal,
+        'I' => :symbols_literal,
+        'w' => :qwords_literal,
+        'W' => :words_literal
+      }
+
+      str_a = [
+        'TEST = %{type}[A B C]',
```

Comment:
Testing each delimiter type seems unnecessary. There is no separate code path for each delimiter in YARD's codebase, so this is a detail of the underlying library and doesn't need testing. In other words, testing for a single delimiter should be sufficient-- if a separate delimiter fails, that would not be a bug with this patch (there is no logic for delimiters in the parser).